### PR TITLE
Platform Docs: Add a documentation page explaining how to use the block library

### PR DESCRIPTION
**Pull request by youknowriad**
Related #53874

## What?

This PR documents the block library package and how to leverage it in third-party block editors. I think it's an interesting page because it highlights some of the current shortcomings that we might want to address separately (Ties to WordPress, boilerplate).

**Note** I have personally tested all the code examples in a third-party vite install.

### Test the documentation website

```
cd platform-docs
npm install
npm start
```

### CODE REVIEW

File: platform-docs/docs/basic-concepts/block-library.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+sidebar_position: 6
+---
+
+# Block Library
+
+The block editor relies on a registry of block types in order to render and edit blocks. The `@wordpress/block-library` package provides a set of core blocks that you can register in your application.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
The block editor relies on a registry of block types to render and edit blocks. The `@wordpress/block-library` package provides a set of core blocks that you can register in your application.
```

Nit: Small tweak to make more concise.

File: platform-docs/docs/basic-concepts/block-library.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+## Registring all block types
+
+Registering blocks require both loading the JavaScript code that make the block type available for use and including the corresponding stylesheets.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
Registering blocks requires both loading the JavaScript code that makes the block type available for use and including the corresponding stylesheets.
```

Small typos.

File: platform-docs/docs/basic-concepts/block-library.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+```js
+import "@wordpress/block-library/build-style/common.css";
+import "@wordpress/block-library/build-style/style.css";
+import "@wordpress/block-library/build-style/editor.css";
+```
+
+## Registering individual blocks
+
+That said, by default the block library includes a very big number of blocks and some of them may contain some WordPress specific logic. For this reason, if you're building a third-party block editor, it's recommended to only register the blocks that you need.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
That said, by default the block library includes a very big number of blocks and some of them may contain some WordPress-specific logic. For this reason, if you're building a third-party block editor, it's recommended to only register the blocks that you need.
```

Nit: Needs a hyphen

File: platform-docs/docs/basic-concepts/block-library.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+The main block type that almost all block editors need is the paragraph block. You can register it with the following code:
+
+```js
+import '@wordpress/block-library/build-module/paragraph/init';
+import '@wordpress/block-library/build-style/paragraph/style.css';
+import '@wordpress/block-library/build-style/paragraph/editor.css';
+```
+
+Also, the paragraph block is often used as the "default block" in the block editor. The default block have multiple purposes:
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
Also, the paragraph block is often used as the "default block" in the block editor. The default block has multiple purposes:
```

Typo

File: platform-docs/docs/basic-concepts/block-library.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+ - **Table block**: `table`
+ - **Verse block**: `verse`
+ - **Separator block**: `separator`
+ - **Spacer block**: `spacer`
+ - **Columns block**: `columns` and `column`
+ - **Group block**: `group`
+ - **Button block**: `buttons` and `button`
+ - **Social links block**: `social-links` and `social-link`
+
+For each block, you'll need to load the JavaScript code and the stylesheets, some blocks have two stylesheets (`style.css` and `editor.css`). For example, to register the heading block, you can use the following code:
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
For each block, you'll need to load the JavaScript code and stylesheets. Some blocks have two stylesheets (`style.css` and `editor.css`). For example, to register the heading block, you can use the following code:
```

Small wording and punctuation tweaks.

File: platform-docs/docs/basic-concepts/block-library.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+```js
+import { setDefaultBlockName } from '@wordpress/blocks';
+
+setDefaultBlockName( 'core/paragraph' );
+```
+
+### The HTML block type
+
+Another important block type that most block editors would want to use is the HTML block. This block allows users to insert arbitrary HTML code in the block editor. It's often used to insert embeds or external content.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
Another important block type that most block editors would want to use is the HTML block. This block allows users to insert arbitrary HTML code in the block editor and is often used to insert embeds or external content.
```

Potential small tweak to reduce impact of consecutive sentences beginning with "it is used".

File: platform-docs/docs/basic-concepts/block-library.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+It is also used by the block editor to render blocks that are not registered in the block editor or as a fallback block type for random HTML content that can't be parsed properly into blocks.
```
Comment (aaronrobertshaw) [inline platform-docs/docs/basic-concepts/block-library.md]:
```suggestion
It is also used by the block editor to render blocks that are not registered in the block editor or as a fallback block type for random HTML content that can't be properly parsed into blocks.
```

Nit: The adverb `properly` needs to be between the two verbs `be` and `parsed`

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for continuing these documentation efforts 👍 

Looking good so far. I've left a few possible tweaks and typo fixes as inline comments.

**Review by talldan — APPROVED** [review]
LGTM!

**Pull request merged**